ada_feeding_perception/ada_feeding_perception/food_on_fork.py

```diff
@@ -0,0 +1,136 @@
+
+        # initialize the model
+        self.model = joblib.load("/home/atharva2/atharvak_node_ws/src/ada_feeding/ada_feeding_perception/ada_feeding_perception/logistic_reg_model.pkl")
+
+    def listener_callback_depth(self, depth_img_msg):
+        """
+        Calculates and publishes probability indicating whether the provided depth image has food on fork
+
+        Parameters:
+            depth_img_msg: depth image message from the subscriber
+        """
+        depth_img = None
+        try:
+            depth_img = self.bridge.imgmsg_to_cv2(depth_img_msg, "passthrough")
+        except CvBridgeError as e:
+            print(e)
+
+        if depth_img is not None:
+            num_pixels = self.food_on_fork_num_pixels(depth_img)
+            # print("num pixels: ", num_pixels)
+            prediction = self.predict_food_on_fork(num_pixels)
+            # print("prediction: ", prediction)
+            float32msg = Float32()
+            float32msg.data = prediction
+            self.publisher_depth.publish(float32msg)
+
+    def food_on_fork_num_pixels(self, depth_img,
+                                left_top_corner=(297, 248),
+                                right_bottom_corner=(422, 332),
+                                min_dist=(330 - 20),
+                                max_dist=(330 + 40)) -> int:
+        """
+        Calculates the number of pixels in the provided depth image (through a depth image message)
+
+        Parameters:
+            depth_img: depth image
+            left_top_corner: Tuple(int, int): Top-left point of the bounding box rectangle
+            right_bottom_corner: Tuple(int, int): Bottom-right point of the bounding box of the rectangle
+            min_dist: int: minimum depth to consider (note that 330 is approx distance to the fork tine)
+            max_dist: int: maximum depth to consider (note that 330 is approx distance to the fork tine)
+
+        Returns:
+            number of pixels within the specified parameter range
+        """
+
+        # consider the points for the rectangle
+        pt1_col, pt1_row = left_top_corner
+        pt2_col, pt2_row = right_bottom_corner
+
+        # create mask that satisfies the rectangle and distance conditions
+        mask_img = np.zeros_like(depth_img, dtype=bool)
+        mask_img[pt1_row:pt2_row, pt1_col:pt2_col] = True
+        mask_img[np.logical_not((min_dist < depth_img) & (depth_img < max_dist))] = False
+
+        return np.count_nonzero(mask_img)
+
+    def predict_food_on_fork(self, num_pixels) -> float:
+        """
+        Calculates the probability of the presence of food on fork
+
+        Parameters:
+            num_pixels: number of pixels detected based on which the probability is output
+
+        Returns:
+            probability of the presence of food on fork
+        """
+        # print("num pixels in method: ", num_pixels)
+        num_pixels = np.array([[num_pixels]])
+        num_pixels_reshape = num_pixels.reshape(-1, 1)
```

Comment:
Why do you have to reshape? You already have a 1x1 array even before reshaping

Reply:
Yup! You are absolutely right! In the previous version, I had a multiple values since we were trying to train/test to get the metrics. But, when I adapted it to this version, I forgot to change it. I have gone ahead and made that change. Thank you for catching!